File: bindings/java-jni/src/main/java/org/maplibre/nativejni/render/OpenGLOwnedTextureDescriptor.java
package org.maplibre.nativejni.render;

import java.util.Objects;

/** Mutable descriptor for OpenGL session-owned texture render targets. */
public final class OpenGLOwnedTextureDescriptor {
  private RenderTargetExtent extent = new RenderTargetExtent();
  private OpenGLContextDescriptor context = new WglContextDescriptor();

  public RenderTargetExtent extent() {
    return extent;
  }

  public OpenGLOwnedTextureDescriptor extent(RenderTargetExtent extent) {
    this.extent = Objects.requireNonNull(extent, "extent");
    return this;
  }

  public OpenGLContextDescriptor context() {
    return context;
  }

  public OpenGLOwnedTextureDescriptor context(OpenGLContextDescriptor context) {
    this.context = Objects.requireNonNull(context, "context");
    return this;
  }
}

File: bindings/kotlin/src/commonMain/kotlin/org/maplibre/nativeffi/render/OpenGLOwnedTextureDescriptor.kt
package org.maplibre.nativeffi.render

/** Mutable descriptor for OpenGL session-owned texture render targets. */
public class OpenGLOwnedTextureDescriptor(
  extent: RenderTargetExtent,
  context: OpenGLContextDescriptor,
) {
  public var extent: RenderTargetExtent = extent

  public var context: OpenGLContextDescriptor = context
}
